src/helpers/useFocusItemHelper.ts

```diff
@@ -0,0 +1,171 @@
+  /**
+   * The number of items
+   */
+  itemsCount: number;
+  /**
+   * The selected item's index
+   */
+  selectedIndex?: number;
+  /**
+   * Where would the item be located (default to CENTER)
+   */
+  offsetType?: OffsetType;
+  /**
+   * Add a margin to the offset (default to true)
+   * This gives a better UX
+   * Not relevant to OffsetType.CENTER
+   */
+  addOffsetMargin?: boolean;
+  /**
+   * How much space (padding \ margin) is there on he left\right of the items
+   */
+  sideSpacing?: number;
+  /**
+   * How much space (padding \ margin) is there between each item
+   */
+  innerSpacing?: number;
+};
+
+// TODO: this is what I want, is there a better way to do it?
+type Offsets = {
+  CENTER: number[];
+  LEFT: number[];
+  RIGHT: number[];
+};
+
+export type ResultProps = {
+  /**
+   * This should be called by each ot the items' onLayout
+   */
+  onItemLayout: (event: LayoutChangeEvent, index: number) => void;
+  /**
+   * The items' width
+   */
+  itemsWidths: number[];
+  /**
+   * Use in order to focus the item with the specified index
+   */
+  focusIndex: (index: number, animated?: boolean) => void;
+};
+
+const useFocusItemsHelper = (props: Props): ResultProps => {
+  const {
+    scrollViewRef,
+    itemsCount,
+    selectedIndex,
+    offsetType = OffsetType.CENTER,
+    addOffsetMargin = true,
+    sideSpacing = 0,
+    innerSpacing = 0
+  } = props;
+  const itemsWidths = useRef<(number | null)[]>(_.times(itemsCount, () => null));
+  const currentIndex = useRef<number>(selectedIndex || 0);
+  const [offsets, setOffsets] = useState<Offsets>({CENTER: [], LEFT: [], RIGHT: []});
+
+  // TODO: reset?
+  //   useEffect(() => {
+  //     itemsWidths.current = _.times(itemsCount, () => null);
+  //   }, [itemsCount]);
+
+  // const contentWidth = _.sum(itemsWidths);
+  // TODO: const scrollEnabled = contentWidth.current > containerWidth;
+
+  const setSnapBreakpoints = useCallback((itemsWidths: number[]) => {
+    if (_.isEmpty(itemsWidths)) {
+      return;
+    }
+
+    const screenCenter = Constants.screenWidth / 2; // TODO: change to something more dynamic?
+    let index = 0;
+    const centeredOffsets = [];
+    let currentCenterOffset = sideSpacing;
+    const leftOffsets = [];
+    leftOffsets.push(0);
+    const rightOffsets = [];
+    rightOffsets.push(-Constants.screenWidth + itemsWidths[0]);
+    while (index < itemsCount) {
+      centeredOffsets[index] = currentCenterOffset - screenCenter + itemsWidths[index] / 2;
+      ++index;
+      currentCenterOffset += itemsWidths[index - 1] + innerSpacing;
+      leftOffsets[index] = leftOffsets[index - 1] + itemsWidths[index - 1];
+      rightOffsets[index] = rightOffsets[index - 1] + itemsWidths[index];
+    }
+
+    if (addOffsetMargin) {
+      index = 1;
+      while (index < itemsCount - 1) {
+        leftOffsets[index] -= itemsWidths[index - 1];
+        rightOffsets[index] += itemsWidths[index + 1];
+        ++index;
+      }
+    }
```

Comment:
Can you explain a little what this does?
Is it for fixing the focusing so the items won't be too near to the edges?

Reply:
Yes, it is for fixing the margin, it gives a margin of the previous' item's width.